### PR TITLE
ablate chesshack

### PR DESCRIPTION
**Pull request by makslevental**
This PR removes some of "sedding" in aiecc because it's not needed since chess caught up (somewhat).

Note, the bottom commit is that ablation and the remainder are refactors I've been sitting on for the past week.

### CODE REVIEW

File: python/compiler/aiecc/configure.py.in [hunk 1]
```diff
@@ -17,6 +17,21 @@ aie_unified_compile = True
 host_disable_compile = not pythonize_bool("@AIECC_HOST_COMPILE@")
 host_architecture = "@LLVM_HOST_TRIPLE@"
 
+peano_install_dir = "@PEANO_INSTALL_DIR@"
+if peano_install_dir == "":
+    peano_install_dir = "peano_not_found"
```
Comment (stephenneuendorffer) [inline python/compiler/aiecc/configure.py.in:22]: These lines seem redundant?

**Review by stephenneuendorffer — APPROVED** [review]
Nice cleanup :)

File: python/compiler/aiecc/configure.py.in (hunk 1 above)
Comment (makslevental) [reply python/compiler/aiecc/configure.py.in:22]: I guess I wasn't sure what `os.path.exists("")` would return. Double checking now it does return `false` so redundant is correct!

**Comment by makslevental** [comment]
@stephenneuendorffer 

This refactoring has uncovered a bug: there is an attribute on `CoreOp` called `link_with` (that was [introduced](https://github.com/Xilinx/mlir-aie/blob/766f1bca304d5b4ebe57b42e5359bf46d36ceeb7/include/aie/Dialect/AIE/IR/AIEOps.td#L293) by @AndraBisca about a month ago) whose purpose is to connect cores with ["kernel objects"](https://github.com/Xilinx/mlir-aie/blob/1028fcdc3fe234be37a063d3f47dcbd2682a593e/reference_designs/ipu-xrt/vision_pipelines/passthrough/aie2_lineBased_8b_1080.mlir#L47):

```mlir
        %core02 = aie.core(%tile02) {
        ...
        } { link_with="passThrough.cc.o" } // indicate kernel object name used by this core
```

**but this attribute is nowhere read during compiling/linking**. Instead `aie-translate --aie-generate-corelist` looks for an [`elf_file` attribute](https://github.com/Xilinx/mlir-aie/blob/196f4acc841567d3d60c793eef89e3cb2c67c9df/lib/Targets/AIETargets.cpp#L224-L223) **which is not now and never has been an `ins` of `CoreOp`** but is present in some of the tests and is mentioned in the doc string for `CoreOp`. 

**Assumption**: a kernel object is an elf file?

Using the `link_with` attributes causes new test fails for tests that ostensibly link against some elf files that aren't created during the test.

**Pull request merged**